src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -670,6 +670,28 @@ private void FlushPendingOutput()
                         }
                     }
 
+                    //create the file in the main thread and flush the contents in the background thread
+                    // In case , there is an error in file creation, the logic would return from here and
+                    // not consume the exception
+                    string baseDirectory = Path.GetDirectoryName(transcript.Path);
+                    
+                    if(Directory.Exists(transcript.Path) || (null != baseDirectory && baseDirectory.Equals(transcript.Path.TrimEnd(System.IO.Path.DirectorySeparatorChar))))
+                    {
+                        Exception exception = PSTraceSource.NewArgumentException(transcript.Path,
+                            InternalHostUserInterfaceStrings.InvalidTranscriptFilePath, 
+                            transcript.Path);
```

Comment:
This `NewArgumentException` override is wrong to use here. Please take a closer look at the implementation of `PSTraceSource.NewArgumentException`.